pkg/bootstrap/cni.go

```diff
@@ -20,21 +20,28 @@ import (
 
 // InstallCNI makes sure all Istio CNI resources have been created.  CRDs are located from
 // files in controller.HelmDir/istio-init/files
-func InstallCNI(ctx context.Context, cl client.Client, config cni.Config) error {
+func InstallCNI(ctx context.Context, cl client.Client, config cni.Config, instanceVersion versions.Version) error {
+	// instanceVersion is from a SMCP spec version
+	if instanceVersion == nil {
+		instanceVersion = versions.DefaultVersion.Version()
+	}
```

Comment:
Is this necessary? If it is, why not just do this in one place (for example, `internalRenderCNI()`)? But I'd prefer to completely remove this and instead have the code fail. This makes it easier to reason about the code (you either pass the version when you invoke the function, or you don't... if the function uses a default value when the caller passes in nil, it's possible this will eventually lead to some weird bug).